src/topp/SageAdapter.cpp

```diff
@@ -487,6 +488,26 @@ class TOPPSageAdapter :
       filenames,
       decoy_prefix);
 
+    // rename SAGE subscores to have prefix "SAGE:"
+    for (auto& id : peptide_identifications)
+    {
+      auto& hits = id.getHits();
+      for (auto& h : hits)
+      {
+        for (const auto meta : extra_scores)
+        {
+          if (h.metaValueExists(meta)
+          {
+            h.setMetaValue("SAGE:" + meta, h.getMetaValue(meta));
+            h.removeMetaValue(meta);        
+          }          
+        }
+      }
+    }
+
+    // remove hits without charge state assigned (fix for downstream bugs)
```

Comment:
I'm honestly wondering. How can a hit not have a charge state. Isn't having a charge state required for matching to precursor masses??

Reply:
I did not check the sage code... maybe it got a good fragment index match? Or charge 1 is always in the index?

Reply:
Maybe we should update sage such that we can specify the range to be sure? Not sure if there was a new release already.

Reply:
Almost ready for new release... just gotta do the final checks on large scale datasets, then you will be able to specify charge state range manually (will try and get it out tonight).

It's not possible that there is a hit without an assigned charge state (short of there being a soundness bug in Rust where we have unsigned integers that are capable of being null 😁). By default, charge states z=2, 3, 4 will be searched if the charge state is not annotated in the mzML file. As to Timo's question, charge z=1 is always present in the fragment index, and in fact is the only charge state present in the index.

It *is* possible that there is a bug (?) in the .pin file writer where this is being omitted somehow. If that's the case, please share an example.

Reply:
@lazear the pin file contains rows where z=other is 1.
So this means that in the mzML charge 1 must be annotated.
Do you report and consider this a valid hit after the update?

Reply:
z=other would imply a charge state of 1 or 7+. Sage only does a sweep through multiple charge states (2,3,4 by default) if (a) there is no charge state for that scan in the mzML or (b) "wide_window" mode is used.

Any rows with z=other won't be affected by the update since they have charge states in the mzML.